### ISSUE TITLE
[BUG] Since version 2.1.0p5 modules not working anymore

### ISSUE THREAD

**Issue by elwood218** [tribe29/ansible-collection-tribe29.checkmk#78]
**Describe the bug**

Folders cannot be created because of changes in the API since 2.1.0p5

```
fatal: [Host1 -> localhost]: FAILED! => changed=false
  msg: 'Error calling API. HTTP code 400. Details: b''{"title": "Bad Request", "status": 400, "detail": "These fields have problems: parent", "fields": {"parent": ["The folder \''/blub/blub/bla\'' could not be found."]}}'', '
```

**Component Name**

`folder`, maybe also `host` 

**Ansible Version**

```console
$ ansible --version
ansible [core 2.12.5]

```

**Checkmk Version**

```console
2.1.0p5 (CEE) and 2.1.0p6
```

**Collection Version**

```console
$ ansible-galaxy collection list
tribe29.checkmk               0.5.0
```
**Environment**

**To Reproduce**

```
- name: "Create site folder"
  tribe29.checkmk.folder:
    server_url: "https://{{ cmk_central }}/"
    site: "{{ cmk_central_site }}"
    automation_user: "{{ cmk_site_user }}"
    automation_secret: "{{ cmk_site_password }}"
    attributes:
      labels:
        test: "test123"
    path: "/BLUB/blub/BLA/FOO"
    title: "Foo"
    state: "present"
  run_once: true
  delegate_to: localhost
  become: no
  notify:
    - activate changes
 ```

**Expected behavior**

**Actual behavior**

**Screenshots**

**Additional context**

Please remove the `.lower()` in following line on folder.py. Please also check host.py.

```
ba36a79b (Lars Getwan    2022-03-11 14:29:03 +0100 125)     return str(p.parent).lower(), p.name
```

**Comment by kain88-de** [comment]
Sure this only happens since p5? I did change the path handling in p4. I also just checked the tests I added and found that I generate paths in a loop. So to get "/Blub/Baa" I first create "/Blub" and then "/Blub/Baa". The error also says that the REST API only wants to create the last folder given. Or in other words. This behaves like `mkdir` and not `mkdir -p`. So my assumption is this has happened before.

**Comment by kain88-de** [comment]
The error message you have written says that the parent folder `/blub/blub/bla` does not exist. Since the rest endpoint does not create parent folders this message is expected. We can fix this in the rest api or let the ansible module deal with that.

**Comment by elwood218** [comment]
Ok.. I have copied the module and fixed it locally by myself anyway. So just take it as notice and do what you think is ok..

The parent folder is already existing and its folder ID (not title!) is: `~BLUB~blub~BLA`
The module does a `.lower()` and then the the ID of the folder is like in the output and it searches for the id `~bla~blub~bla` and does not find it.

The `.lower()` is since march in the module so anything real strange was done by the API which did not throw this error and it was working like expected and folders got created. But since p5 ist was not working anymore and I debugged it several hours before I created this issue.

**Comment by elwood218** [comment]
Btw.. and just fyi - I also fixed the host module which was not working anymore. The API Maybe was sending not the first `/` before and now it does.. so to fix the host module just remove the slash which is  set by the module:

OLD:     
```
current_folder = "/%s" % extensions.get("folder", "")
```

NEW:
```
current_folder = "%s" % extensions.get("folder", "")
```

**Comment by robin-tribe29** [comment]
@elwood218 your issue should be fixed with #147. It will be included in version `0.9.0` which will be released today.
Can you verify that the fix works for you too and close this issue if so?

**Comment by robin-tribe29** [comment]
Closing this due to staleness, as there is another fix in 993b4ce7a3957d881389f7f9216e828e7bebefc0.

If there is anything else, please open a new issue.
